### PR TITLE
Issue#1195 Improve Pool Delete UX

### PR DESCRIPTION
**Pull request by gkadillak**
## Deletion of Pool in Modal

Users are now able to delete a pool from the pool detail page via a force delete checkbox. The size of the share is displayed (rounded to the nearest integer) along with a warning saying that all associated information will be removed.

Also included is a fix where the delete button doesn't show up for the root pool in the pool detail page.

image

### CODE REVIEW

**Comment by gkadillak** [comment]
Not a problem, @MFlyer! Would that be worth adding to the community contribution docs?

**Comment by MFlyer** [comment]
> Would that be worth adding to the community contribution docs?

That would be a good thing!

**Comment by gkadillak** [comment]
Ok I'll make a pull request to the docs as well with that change :)

**Comment by gkadillak** [comment]
@MFlyer take a look and tell me what you think!

**Comment by sfranzen** [comment]
@MFlyer So our "official" JS/HTML coding style is going to be 4 spaces per level? I've been using 2 but don't mind changing it in future PRs.

**Comment by MFlyer** [comment]
Hi @sfranzen,
I assumed a 4 spaces tab looking to other Rockstor files and @phillxnet coding too (4 spaces on python, 4 spaces on js, same indentation).
Do we agree on this?
I think @schakrava should have the last say.

**Comment by sfranzen** [comment]
@MFlyer,
Yes, you're right, except for the couple of dashboard widget views apparently. Since I started by copy/pasting some code together from there I was looking at my files and just seeing 2 spaces. Going with 4 now! :+1:

File: src/rockstor/storageadmin/static/storageadmin/js/templates/pool/pool_details_layout.jst [hunk 1]
```diff
@@ -104,10 +108,28 @@
...
+<div id="delete-pool-modal" class="modal fade" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+        <h3 id="myModalLabel">Delete Pool</h3>
+      </div>
+      <div class="modal-body">
+        <div class="messages"></div>
+        <h4>Pool {{pool.name}} will be deleted along with the following shares (and all information associated with shares including Snapshots, NFSExports, Samba configs, SFTP, Replics). Are you sure?</h4>
```
Comment (sfranzen) [inline src/rockstor/storageadmin/static/storageadmin/js/templates/pool/pool_details_layout.jst]: Spelling: replics -> replicas

**Comment by schakrava** [comment]
Yes, 4 spaces guys!

**Comment by gkadillak** [comment]
I've seen 2 spaces work well with js but I'm impartial.

**Comment by MFlyer** [comment]
My personal opinion:
I'm impartial like @gkadillak , 1+ on 4 spaces indent because of better reading.

@phillxnet @priyaganti ?

**Comment by priyaganti** [comment]
I am not sure about the spaces to be used but I have tested your changes. The force delete feature looks good and works as per the user request. Thanks @gkadillak for taking the initiative.

**Comment by gkadillak** [comment]
Happy to hear that @priyaganti ! Thanks for giving it a shot :)

**Comment by schakrava** [comment]
Hey @gkadillak. Nicely done. I just tested your changes and it all looks good. I just have 2 comments.
1. The behavior of delete button in the pools table view is still the old one. Please update it. See if you can reuse the same template/modal for both views.
2. Update the indentation to 4 spaces while you are at it :)

**Comment by gkadillak** [comment]
4 spaces it is! I'll update the table view behavior as well.

**Comment by gkadillak** [comment]
@schakrava finally finished up this work. Can you take a look? Thanks!

**Comment by schakrava** [comment]
Hey @gkadillak, sorry for the delay but just got to test this morning. You'll notice that I pushed to your fork, mostly rebase related stuff to save you time.

It's still not working for me. Both in the details view and table view confirmation modals, list of shares is not shown even when I have one in the Pool. Can you take a second look at that logic?
delete_pool_confirm

**Comment by gkadillak** [comment]
Yep I'll take another look!

**Comment by schakrava** [comment]
Are you able to reproduce the bug after your rebase?

**Comment by gkadillak** [comment]
Unfortunately, no. Is there anything in the console?

**Comment by schakrava** [comment]
Thanks for your work @gkadillak . I had to make a few changes to get it all working, but we are good now.

**Pull request merged**